src/lib/util/dbuff.h

```diff
@@ -410,6 +411,68 @@ static inline ssize_t fr_dbuff_net_from_uint64(fr_dbuff_t *dbuff, uint64_t num)
 }
 #define FR_DBUFF_NET_FROM_UINT64_RETURN(_dbuff, _num) FR_DBUFF_RETURN(fr_dbuff_net_from_uint64, _dbuff, _num)
 
+/** Copy a signed 16-bit integer into a buffer in wire format (big endian)
+ *
+ * @param[in] dbuff	to copy data to
+ * @param[in] num	value to copy
+ * @return
+ * 	0	no data set
+ * 	>0	(2, actually) the number of bytes set in the dbuff
+ * 	<0	the number of bytes required
+ */
+static inline ssize_t fr_dbuff_net_encode_int16(fr_dbuff_t *dbuff, int16_t num)
+{
+	return fr_dbuff_net_encode_uint16(dbuff, (uint16_t) num);
+}
+#define FR_DBUFF_NET_FROM_INT16_RETURN(_dbuff, _num) FR_DBUFF_RETURN(fr_dbuff_net_from_int16, _dbuff, _num)
+
+/** Copy a signed 32-bit integer into a buffer in wire format (big endian)
+ *
+ * @param[in] dbuff	to copy data to
+ * @param[in] num	value to copy
+ * @return
+ * 	0	no data set
+ * 	>0	(4, actually) the number of bytes set in the dbuff
+ * 	<0	the number of bytes required
+ */
+static inline ssize_t fr_dbuff_net_encode_int32(fr_dbuff_t *dbuff, int32_t num)
+{
+	return fr_dbuff_net_encode_uint32(dbuff, (uint32_t) num);
+}
+#define FR_DBUFF_NET_FROM_INT32_RETURN(_dbuff, _num) FR_DBUFF_RETURN(fr_dbuff_net_from_int32, _dbuff, _num)
+
+/** Copy a signed 64-bit integer into a buffer in wire format (big endian)
+ *
+ * @param[in] dbuff	to copy data to
+ * @param[in] num	value to copy
+ * @return
+ * 	0	no data set
+ * 	>0	(8, actually) the number of bytes set in the dbuff
+ * 	<0	the number of bytes required
+ */
+static inline ssize_t fr_dbuff_net_encode_int64(fr_dbuff_t *dbuff, int64_t num)
+{
+	return fr_dbuff_net_encode_uint64(dbuff, (uint64_t) num);
+}
+#define FR_DBUFF_NET_FROM_UINT64_RETURN(_dbuff, _num) FR_DBUFF_RETURN(fr_dbuff_net_from_uint64, _dbuff, _num)
+
+#define fr_dbuff_net_encode(dbuff, value) \
+	_Generic((value), \
+		int16_t		: fr_dbuff_net_encode_int16(dbuff, value), \
+		int32_t		: fr_dbuff_net_encode_int32(dbuff, value), \
+		int64_t		: fr_dbuff_net_encode_int64(dbuff, value), \
+		uint16_t	: fr_dbuff_net_encode_uint16(dbuff, value), \
+		uint32_t	: fr_dbuff_net_encode_uint32(dbuff, value), \
+		uint64_t	: fr_dbuff_net_encode_uint64(dbuff, value) \
```

Comment:
You should cast `value` explicitly to an integer of the correct width to avoid warnings.  `_Generic` is a bit strange in that the compiler appears to evaluate every possible type for a given use.